posthog/models/web_preaggregated/sql.py

```diff
@@ -449,6 +476,9 @@ def WEB_BOUNCES_INSERT_SQL(
                 argMinMerge(raw_sessions.initial_geoip_subdivision_city_name) AS city_name,
                 argMinMerge(raw_sessions.initial_geoip_subdivision_1_code) AS region_code,
                 argMinMerge(raw_sessions.initial_geoip_subdivision_1_name) AS region_name,
+                isNotNull(nullIf(nullIf(argMinMerge(raw_sessions.initial_gclid), 'null'), '')) AS has_gclid,
```

Comment:
according to my experience, CH is not very smart about optimizing things like that: `isNotNull(nullIf(nullIf(`
`raw_sessions.initial_gclid IN ('', 'null')` or better `notEmpty(raw_sessions.initial_gclid) AND raw_sessions.initial_gclid != 'null'`

Reply:
Thanks @orian , I will update this, it was complaining that it couldn't guarantee the column data type depending on the expression, so that's why I added more